docs/customization_guide/inference_protocols.md

```diff
@@ -432,3 +432,165 @@ JavaCPP-presets. You can do this using the following steps:
+* "--kafka-producer-topic" : Single, default topic to be used by Triton's producer when "response_topic" is not specified in an "inference_request_header". 
+
+The following sections will outline the Kafka request and response message
+designs. It is intended that the schemas mentioned below be constructed
+using JSON before being converted into a string and sent to the Kafka
+broker. As such, the schemas outlined below will be represented as
+JSON objects for simplicity, where $number and $string refer to the
+fundamental JSON types. #optional indicates an optional JSON field.
+$string($number) refers to an integer value that has been converted
+to a string.
+
+## Kafka Inference Request
+Once the Kafka endpoint is running, Triton will begin consuming inference
+requests from the provided topics. Triton expects inference requests stored
+in the Kafka broker to be comprised of a header and a payload. 
+
+#### Message Header
+The inference request header should have the following format:
+
+```C++
+$inference_request_header
+{
+  "id" : $string($number),
+  "model_name" : $string,
+  "model_version" : $string($number),
+  "payload_schema_length" : $string($number),
+  "response_topic" : $string, //optional
+  "response_key" : $string //optional
+}
+```
+
+* "id" : A unique identifier for the corresponding inference request.
+* "model_name" : The name of the model to perform the inference.
+* "model_version" : The version of the model to perform the inference.
+* "payload_schema_length" : The length of the $payload_request_schema.
+* "response_topic" : The topic Triton's producer should send the response message
+to.
+* "response_key" : The key Triton's producer should use when sending a response
+message.
+
+#### Message Payload
+The expected payload of the inference request is composed of two parts:
+schema and inference data (as raw bytes) concatenated together. The payload
+schema should have the following format:
+
+```C++
+$payload_request_schema
+{
+  "inputs" : [ $request_input, ... ],
+  "outputs" : [ $request_output, ... ]
+}
+```
+Where a $request_input is defined as:
+```C++
+$request_input
+{
+  "name" : $string,
+  "shape" : $string([ $number, ... ]),
+  "datatype"  : $string,
+  "parameters" : {"binary_data_size":$string(int)}
+}
+```
+* "name" : The name of the input tensor.
+* "shape" : The shape of the input tensor.
+* "datatype" : The data-type of the input tensor, defined in [Datatypes](https://github.com/triton-inference-server/server/blob/main/docs/model_configuration.md#datatypes).
+* "parameters" : Currently only supporting "binary_data_size" which expects the size of the input tensor data in bytes.
+
+And a $request_output is defined as:
+```C++
+request_output
+{
+  "name" : $string
+  "parameters" : {"binary_data":true}
+}
+```
+* "name" : The name of the output tensor.
+* "parameters" : Currently only supporting "binary_data", which must be set to true.
```

Comment:
Why user need to set this if we only allow `true`.

Reply:
It was left in for future extensibility for when we do have multiple optional parameters. This way, the clients will already have the infrastructure developed to provide this field.